### PR TITLE
feat(select): accessibility improvement on select and option component

### PR DESCRIPTION
**Pull request by MehmetCanBOZ**
This pull request improves the accessibility of select and option components.
Fixes: https://github.com/Trendyol/baklava/issues/587

### CODE REVIEW

File: src/components/select/bl-select.ts [hunk 1]
```diff
@@ -343,12 +343,14 @@ export default class BlSelect<ValueType extends FormValue = string> extends Form
         'dirty': this.dirty,
       })}
       @keydown=${this.handleKeydown}
+      aria-label="${ifDefined(this.label)}"
```
[the remark this answers is not in the document]
Comment (muratcorlu) [reply src/components/select/bl-select.ts:346]: We also need to use [`aria-multiselectable`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-multiselectable)

**Review by muratcorlu — COMMENTED** [review]
Thanks for the contribution! I have some remarks.

File: src/components/select/bl-select.ts [hunk 2]
```diff
@@ -343,12 +343,14 @@ export default class BlSelect<ValueType extends FormValue = string> extends Form
...
     >
       ${label} ${this.inputTemplate()}
       <div
         class="popover"
         tabindex="${ifDefined(this._isPopoverOpen ? undefined : '-1')}"
         @bl-select-option=${this._handleSelectOptionEvent}
+        role="list-box"
```
[the remark this answers is not in the document]
Comment (MehmetCanBOZ) [reply src/components/select/bl-select.ts]: I fixed it, thank you.

File: src/components/select/bl-select.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (MehmetCanBOZ) [reply src/components/select/bl-select.ts]: I fixed it, thank you. Could you review it again, if you are available?

File: src/components/select/bl-select.ts [hunk 3]
```diff
@@ -343,6 +343,9 @@ export default class BlSelect<ValueType extends FormValue = string> extends Form
         'dirty': this.dirty,
       })}
       @keydown=${this.handleKeydown}
+      aria-labelledby="label"
+      role="list-box"
+      aria-multiselectable="true"
```
Comment (muratcorlu) [inline src/components/select/bl-select.ts:348]: This should be conditional. Like: `aria-multiselectable="${this.multiple}"`

**Comment by MehmetCanBOZ** [comment]
Yes, you are correct. I tried several options for the role attribute with VoiceOver, and the best solution that I found is to give role=“button", aria-haspopup=“listbox”  and aria-expanded="{isOpened}" attribute to the fieldset and  role=“listbox” attribute for class=“popover” element. Maybe you can try again with a screen reader and give feedback about it.

**Review by muratcorlu — APPROVED** [review]
I checked with voiceover and it looks good to me. Thank you!

**Pull request merged**